src/parser/mod.rs

```diff
@@ -7972,10 +7972,13 @@ impl<'a> Parser<'a> {
                     _ => {}
                 }
 
+                // only allow to use TABLE keyword for DESC|DESCRIBE statement
+                let has_table_keyword = self.parse_keyword(Keyword::TABLE);
```

Comment:
I didn't add the dialect guard since the `TABLE` keyword is allowed in multiple dialects(Snowflake and ClicHouse), and it should be good to let the downstream determine if it's a valid syntax.

Reply:
See https://github.com/sqlparser-rs/sqlparser-rs/issues/1385